### PR TITLE
MM-56090 Group Search fix

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary

In the group search, the name is parsed on  ['.', '-', '_'] and each word is filtered if it STARTS_WITH the filter. The unparsed display name is also searched. When creating a group, spaces in the display name are removed. Therefore a group "ABC Team" creates the name "abcteam". And without a deliminator only ['abcteam', 'abc team'] are filtered via STARTS_WITH, so a filter of "team" would not be found. 

Update group search to work the same as it does with at_mentions. That is, parse the display name on space(' ') and add each word of the display name to the search so the above would search for STARTS_WITH on the following words.
['abcteam', 'abc', 'team' ,'abc team']

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-56090

#### Screenshots

|  before  |  after  |
|----|----|
| Screenshot 2024-02-05 at 3 52 17 PM | Screenshot 2024-02-05 at 3 50 02 PM
 |

Screenshot 2024-02-05 at 3 50 27 PM

#### Release Note

```release-note
Fixed group search to parse display name.
```

### CODE REVIEW

File: webapp/channels/src/packages/mattermost-redux/src/utils/group_utils.test.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ describe('group utils', () => {
         };
         const groupC = {
             id: 'groupid3',
-            name: 'software-engineers',
+            name: 'softwareengineers',
```
Comment (sbishel) [inline webapp/channels/src/packages/mattermost-redux/src/utils/group_utils.test.ts:42]: When creating via the UI, the spaces are removed, not replaced.

File: webapp/channels/src/packages/mattermost-redux/src/utils/group_utils.test.ts [hunk 2]
```diff
@@ -9,9 +9,9 @@ describe('group utils', () => {
     describe('filterGroupsMatchingTerm', () => {
         const groupA = {
             id: 'groupid1',
-            name: 'board-group',
+            name: 'board.group',
```
Comment (sbishel) [inline webapp/channels/src/packages/mattermost-redux/src/utils/group_utils.test.ts:12]: updated to test a different separator.

**Review by hmhealey — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - this is working as expected 👍

**Pull request merged**